### PR TITLE
Improve detection of click event handler

### LINKED ISSUE #503: DCC orders randomly failing (505)
### Describe the Bug

This issue is about cases where the PayPal Card Processing would fail.

Following update 1.7.0, we got a new error message that occurs when the behavior occurs:
`No PayPal order found in the current WooCommerce session`

The above error message is displayed because for some reason “Place order” button does not call create order endpoint but WC process checkout.

It seems like switching other payment gateways and then switching back to PayPal Card Processing results in some kind of reload, and then our event listeners are not added (https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js#L125) because `formSubmissionSubscribed` is true.

### To Reproduce
1. Enable PayPal Card Processing
2. Enable other payment gateways
3. Add product to cart, visit checkout
4. Switch over some payment gateways
5. Switch back to PayPal Card Processing gateway 
6. Input credit card details and hit Place order

### Expected Behavior
Order processes properly.

### Actual Behavior
Order instantly fails with error above.

### Environment
 - WordPress Version: 5.9
 - WooCommerce Version: 6.2
 - Plugin Version: 1.7.0-test2

### PR DESCRIPTION
**Pull request by Dinamiko**
Fixes #503.

### CODE REVIEW

File: modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js [hunk 1]
```diff
@@ -122,16 +122,18 @@ class CreditCardRenderer {
...
-                document.querySelector(wrapper + ' button').addEventListener(
-                    'click',
-                    event => {
-                        event.preventDefault();
-                        this._submit(contextConfig);
-                    }
-                );
-                this.formSubmissionSubscribed = true;
-            }
+           const checkoutForm = document.forms.checkout;
```
Comment (AlexP11223) [inline modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js]: hm, does it work with mollie now? I thought we would need to add the attribute to something that gets reloaded during this switch (that is the card form).

File: modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js [hunk 2]
```diff
@@ -122,16 +122,18 @@ class CreditCardRenderer {
...
+           if(checkoutForm.getAttribute('data-ppcp-subscribed') !== true) {
```
Comment (AlexP11223) [inline modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js]: missing space

File: modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js [hunk 3]
```diff
@@ -122,16 +122,18 @@ class CreditCardRenderer {
...
 
-            if (!this.formSubmissionSubscribed) {
-                document.querySelector(wrapper + ' button').addEventListener(
-                    'click',
-                    event => {
-                        event.preventDefault();
-                        this._submit(contextConfig);
-                    }
-                );
-                this.formSubmissionSubscribed = true;
```
Comment (AlexP11223) [inline modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js:133]: `this.formSubmissionSubscribed = false;` in constructor can be removed now.

File: modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js (hunk 1 above)
Comment (Dinamiko) [reply modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js]: yup, it works adding the attribute to checkout form. 

I tried adding it to the hosted fields form wrapper (https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js#L75) but it does not work.

It works on the hosted fields button wrapper (https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js#L15) so this is what I used finally.

**Review by AlexP11223 — APPROVED** [review]

**Pull request merged**